### PR TITLE
feat(payment-wallet): CHK-2775 init domain for payment-wallet

### PR DESCRIPTION
**Pull request by infantesimone**
### Motivation and context

The goal of this PR is to init new domain for payment-wallet, in particular a new namespace `pay-wallet` and keyvault in north Italy.
The current [wallet](https://github.com/pagopa/pagopa-infra/tree/main/src/domains/wallet-app) domain will be migrated to this one.

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/pay-wallet-app/05_aks_middleware_tools.tf [hunk 1]
```diff
@@ -0,0 +1,53 @@
+module "tls_checker" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//tls_checker?ref=v6.3.0"
+
+  https_endpoint                         = local.wallet_hostname
```
Comment (pietro-tota) [inline src/domains/pay-wallet-app/05_aks_middleware_tools.tf:4]:
```suggestion
  https_endpoint                         = local.payment_wallet_hostname
```

what about refactoring also those variable names?

File: src/domains/pay-wallet-app/99_locals.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  pagopa_apim_name = "${local.product}-apim"
+  pagopa_apim_rg   = "${local.product}-api-rg"
+  pagopa_apim_snet = "${local.product}-apim-snet"
+
+  pagopa_vnet_integration = "pagopa-${var.env_short}-vnet-integration"
+  pagopa_vnet_rg          = "pagopa-${var.env_short}-vnet-rg"
+
+  apim_hostname   = "api.${var.apim_dns_zone_prefix}.${var.external_domain}"
+  wallet_hostname = "${var.location_short}${var.env}.wallet.internal.${var.env}.platform.pagopa.it"
```
Comment (pietro-tota) [inline src/domains/pay-wallet-app/99_locals.tf:38]:
```suggestion
  wallet_hostname = "${var.location_short}${var.env}.payment-wallet.internal.${var.env}.platform.pagopa.it"
```

This should be payment-wallet, right?

File: src/domains/pay-wallet-app/05_aks_middleware_tools.tf [hunk 3]
```diff
@@ -0,0 +1,53 @@
+module "tls_checker" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//tls_checker?ref=v6.3.0"
+
+  https_endpoint                         = local.payment_wallet_hostname
+  alert_name                             = local.wallet_hostname
```
Comment (pietro-tota) [inline src/domains/pay-wallet-app/05_aks_middleware_tools.tf:5]: sorry, my bad, in previous suggestion I linked only one variable, `wallet_hostname` to `payment_wallet_hostname` should be applied also here and in locals file, right?

Comment (infantesimone) [reply src/domains/pay-wallet-app/05_aks_middleware_tools.tf]: fixed

**Pull request merged**